src/main/scala/firrtl/Utils.scala

```diff
@@ -177,179 +180,198 @@ object Utils extends LazyLogging {
-      case DoPrim(op, args, consts, _) => "_" + op + (args.map(niceName(depth - 1)) ++ consts.map("_" + _)).mkString("")
-      case Mux(cond, tval, fval, _) if depth <= 0 => "_mux"
-      case Mux(cond, tval, fval, _) => "_mux" + Seq(cond, tval, fval).map(niceName(depth - 1)).mkString("")
-      case UIntLiteral(value, _) => "_" + value
-      case SIntLiteral(value, _) => "_" + value
+      case SubAccess(expr, index, _, _)               => niceName(depth)(expr) + niceName(depth - 1)(index)
+      case SubField(expr, field, _, _)                => niceName(depth)(expr) + "_" + field
+      case SubIndex(expr, index, _, _)                => niceName(depth)(expr) + "_" + index
+      case DoPrim(op, args, consts, _) if depth <= 0  => "_" + op
+      case DoPrim(op, args, consts, _)                => "_" + op + (args.map(niceName(depth - 1)) ++ consts.map("_" + _)).mkString("")
+      case Mux(cond, tval, fval, _) if depth <= 0     => "_mux"
+      case Mux(cond, tval, fval, _)                   => "_mux" + Seq(cond, tval, fval).map(niceName(depth - 1)).mkString("")
+      case UIntLiteral(value, _)                      => "_" + value
+      case SIntLiteral(value, _)                      => "_" + value
     }
   }
+
   /** Maps node name to value */
   type NodeMap = mutable.HashMap[String, Expression]
 
   def isTemp(str: String): Boolean = str.head == '_'
 
   /** Indent the results of [[ir.FirrtlNode.serialize]] */
   @deprecated("Use ther new firrt.ir.Serializer instead.", "FIRRTL 1.4")
-  def indent(str: String) = str replaceAllLiterally ("\n", "\n  ")
-
-  implicit def toWrappedExpression (x:Expression): WrappedExpression = new WrappedExpression(x)
-  def getSIntWidth(s: BigInt): Int = s.bitLength + 1
-  def getUIntWidth(u: BigInt): Int = u.bitLength
-  def dec2string(v: BigDecimal): String = v.underlying().stripTrailingZeros().toPlainString
-  def trim(v: BigDecimal): BigDecimal = BigDecimal(dec2string(v))
-  def max(a: BigInt, b: BigInt): BigInt = if (a >= b) a else b
-  def min(a: BigInt, b: BigInt): BigInt = if (a >= b) b else a
-  def pow_minus_one(a: BigInt, b: BigInt): BigInt = a.pow(b.toInt) - 1
+  def indent(str: String) = str.replaceAllLiterally("\n", "\n  ")
+
+  implicit def toWrappedExpression(x: Expression): WrappedExpression = new WrappedExpression(x)
+  def getSIntWidth(s:                 BigInt):     Int = s.bitLength + 1
+  def getUIntWidth(u:                 BigInt):     Int = u.bitLength
+  def dec2string(v:                   BigDecimal): String = v.underlying().stripTrailingZeros().toPlainString
+  def trim(v:                         BigDecimal): BigDecimal = BigDecimal(dec2string(v))
+  def max(a:                          BigInt, b: BigInt): BigInt = if (a >= b) a else b
+  def min(a:                          BigInt, b:   BigInt): BigInt = if (a >= b) b else a
+  def pow_minus_one(a:                BigInt, b:   BigInt): BigInt = a.pow(b.toInt) - 1
   val BoolType = UIntType(IntWidth(1))
```

Comment:
This one is pretty annoying, I'm all for lining up types but not like this. Perhaps a good idea to add spaces between methods, or at least one after `toWrappedExpression`